ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -51,12 +52,16 @@ class HomeScreenMapContainerViewModel
 @Inject
 internal constructor(
   private val resources: Resources,
-  private val surveyRepository: SurveyRepository,
+  private val mapStateRepository: MapStateRepository,
   private val locationOfInterestRepository: LocationOfInterestRepository,
   private val locationController: LocationController,
   private val mapController: MapController,
+  surveyRepository: SurveyRepository,
   offlineAreaRepository: OfflineAreaRepository
 ) : BaseMapViewModel(locationController, mapController) {
+
+  private var activeSurveyId: String = ""
```

Comment:
Why are we maintaining this state locally? Shouldn't we fetch the value from our source of truth (i.e. LocalValueStore)?

Reply:
UI state may lag behind the source of truth in the local datastore; in this case, we're using the UIs understanding of the current survey instead of the local stores since that is what the camera position is referring not. This race condition is unlikely cause issues in practice, but the UI self-reporting the context for camera moves seems to be more correct.

Reply:
Thanks for clarifying.